api4/user.go

```diff
@@ -149,6 +162,17 @@ func getUserByUsername(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	userTermsOfService, err := c.App.GetUserTermsOfService(user.Id)
```

Comment:
You need the `if c.IsSystemAdmin() || c.App.Session.UserId == user.Id {` check here too I think?

Reply:
Done.